subsys/net/ip/6lo.c

```diff
@@ -756,7 +756,7 @@ static inline bool compress_IPHC_header(struct net_pkt *pkt,
 
 	/* UDP header compression */
 	if (IS_ENABLED(CONFIG_NET_UDP)) {
-		struct net_udp_hdr hdr, *udp;
+		struct net_udp_hdr hdr __unused, *udp;
```

Comment:
This works but I suggest to fix this other way.
Please edit `include/net/udp.h` and change `net_udp_get_hdr()` macro to
`static inline struct net_udp_hdr *net_udp_get_hdr(struct net_pkt *pkt, struct net_udp_hdr *hdr) { return NULL; }`
and similar change to net_udp_set_hdr(). This way we do not need to add these __unused strings around.